src/main/scala/sbtsonar/SonarPlugin.scala

```diff
@@ -49,29 +52,42 @@ object SonarPlugin extends AutoPlugin {
     sonarScan := {
       implicit val logger: Logger = streams.value.log
 
-      val sonarHome = sys.env
-        .get("SONAR_SCANNER_HOME")
-        .orElse(sys.props.get("sonarScanner.home"))
-        .getOrElse(
-          sys.error(
-            "SONAR_SCANNER_HOME environment variable or sonarScanner.home system property not defined."
-          )
-        )
-
-      // Update the external properties file if the sonarUseExternalConfig is set to true.
-      if (sonarUseExternalConfig.value)
-        updatePropertiesFile(baseDirectory.value, SonarExternalConfigFileName, version.value)
-
       //Allow to set sonar properties via system properties: [https://docs.sonarqube.org/display/SONAR/Analysis+Parameters]
       val mergedSonarProperties = sonarProperties.value ++ sys.props.filterKeys(_.startsWith("sonar."))
 
-      val args = sonarScannerArgs(sonarUseExternalConfig.value, mergedSonarProperties, version.value)
+      if (sonarUseExternalConfig.value) {
```

Comment:
It is a good idea to keep the existing functionality and still allow execution of the external `sonar-scanner` as a fallback option in case of any issues 👍, but we shouldn't be linking it to the `sonarUseExternalConfig` setting (we should use a separate setting for this - maybe `fallbackToExternalSonarScanner` defaulting to `false`? I'm open to suggestions.). The `sonarUseExternalConfig` setting is meant to enable you to use the properties from `sonar-project.properties` file and I think we should provide this functionality regardless of whether you're using the external `sonar-scanner` executable or the embedded scanner. In the case of the latter, you will need to load this file and convert the properties into a map so we can merge those with the other `sonar.` system props (they do the same in the sonnar-scanner-cli project https://github.com/SonarSource/sonar-scanner-cli/blob/050dd2d15cf83feb44857ffec5fe0b6073797bf7/src/main/java/org/sonarsource/scanner/cli/Conf.java#L82).